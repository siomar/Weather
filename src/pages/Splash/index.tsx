import React, {useEffect} from 'react';
import {View, StyleSheet} from 'react-native';

interface SplashProps {
  navigation: any;
}

const Splash = ({navigation}: SplashProps) => {
  useEffect(() => {
    setTimeout(() => {
      navigation.replace('Home');
    }, 1000);
  });
  return <View style={styles.bg} />;
};

const styles = StyleSheet.create({
  bg: {
    flex: 1,
    backgroundColor: '#ffcc01',
  },
});

export default Splash;
