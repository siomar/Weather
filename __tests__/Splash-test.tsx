import 'react-native';
import React from 'react';
import Splash from '../src/pages/Splash';

import renderer from 'react-test-renderer';
jest.useFakeTimers();

const tree = renderer.create(<Splash navigation={undefined} />);

describe('Home tests', () => {
  test('create snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
  it('should renders correctly', () => {
    renderer.create(<Splash navigation={undefined} />);
  });
});
