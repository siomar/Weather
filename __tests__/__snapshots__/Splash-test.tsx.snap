// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffcc01",
      "flex": 1,
    }
  }
/>
`;
